libraries/botbuilder-dialogs/src/memory/scopes/settingsMemoryScope.ts

```diff
@@ -205,4 +221,29 @@ export class SettingsMemoryScope extends MemoryScope {
             return result;
         }, {});
     }
+
+    private static filterSettings(settings: Record<string, unknown>): Record<string, unknown> {
+        const result = Object.assign({}, settings);
+        SettingsMemoryScope.blockingList.forEach((path) => SettingsMemoryScope.deletePropertyPath(result, path));
+        return result;
+    }
+
+    private static deletePropertyPath(obj, path: string): void {
+        if (!obj || !path) {
+            return;
+        }
+        const pathArray = path.split(':');
+
+        for (let i = 0; i < pathArray.length - 1; i++) {
+            const realKey = Object.keys(obj).find((key) => key.toLowerCase() === pathArray[i].toLowerCase());
```

Comment:
It's probably worth building an index of lowercase key to resolved key once so you aren't enumerating all the keys multiple times.

```typescript
const lowerToKey = Object.keys(obj).reduce((acc, key) => {
  acc[key.toLowerCase()] = key;
}, {});
...
const realKey = lowerToKey[pathArray[i].toLowerCase()];
```